src/dstack/_internal/core/services/ssh/tunnel.py

```diff
@@ -198,23 +195,26 @@ def open(self) -> None:
             raise SSHError(msg) from e
         if r.returncode == 0:
             return
-        with open(self.log_path, "rb") as f:
-            error = f.read()
-        logger.debug("SSH tunnel failed: %s", error)
-        raise get_ssh_error(error)
+        stderr = self._read_log_file()
+        logger.debug("SSH tunnel failed: %s", stderr)
+        raise get_ssh_error(stderr)
 
     async def aopen(self) -> None:
+        loop = asyncio.get_event_loop()
+        await loop.run_in_executor(None, self._remove_log_file)
```

Comment:
(nit) Consider using `dstack._internal.server.utils.common.run_async`.

I also think we usually perform file operations directly in coroutines, although it is questionable.

Reply:
1. Is it okay to import from `_internal.server` inside `_internal.core`? 
2. Yeah, I was not sure if it's acceptable to run blocking file I/O operations (even if the filesystem is local) inside async methods, but I didn't find neither other usages in the code nor dependencies like `aiofiles`.

cc @r4victor

Reply:
> Is it okay to import from _internal.server inside _internal.core?

It's better not to, so I'd leave the code as is for now. Later, run_async can be moved to core.